src/geometry/ArborX_HyperBox.hpp

```diff
@@ -0,0 +1,116 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_HYPERBOX_HPP
+#define ARBORX_HYPERBOX_HPP
+
+// FIXME_CXX17: remove first clause and update the comment once we require C++17
+#if (__cplusplus < 201703L) || (KOKKOS_VERSION < 30699)
+#error "Using multi-dimensional Box requires C++17 and Kokkos 3.7"
+#endif
+
+#include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
+#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
+#include <ArborX_GeometryTraits.hpp>
+#include <ArborX_HyperPoint.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+namespace ArborX
+{
+
+namespace Experimental
+{
+/**
+ * Axis-Aligned Bounding Box. This is just a thin wrapper around an array of
+ * size 2x spatial dimension with a default constructor to initialize
+ * properly an "empty" box.
+ */
+template <int DIM, class FloatingPoint = float>
+struct Box
+{
+  KOKKOS_FUNCTION
+  constexpr Box()
+  {
+    for (int d = 0; d < DIM; ++d)
+    {
+      _min_corner[d] =
+          KokkosExt::ArithmeticTraits::finite_max<FloatingPoint>::value;
```

Comment:
For `3.7.00` and later `finite_min/max` are available in `Kokkos` directly.

Reply:
It's in `Kokkos::Experimental` though.